mlflow_project/src/rf_test/train.py

```diff
@@ -0,0 +1,120 @@
+import argparse
+from functools import partial
+
+import mlflow
+import mlflow.sklearn
+
+from cuml.metrics.accuracy import accuracy_score as accuracy_score
+from cuml.preprocessing.model_selection import train_test_split as train_test_split
+from cuml.ensemble import RandomForestClassifier as RandomForestClassifier
+
+from hyperopt import fmin, tpe, hp, Trials, STATUS_OK
+
+
+def load_data(fpath):
+    """
+    Simple helper function for loading data to be used by GPU models.
+
+    :param fpath: Path to the data to be ingested
+    :return: DataFrame wrapping the data at [fpath]. Data will be in either a Pandas or RAPIDS (cuDF) DataFrame
+    """
+    import cudf
+
+    df = cudf.read_parquet(fpath)
+
+    return df
+
+
+def _train(params, fpath, hyperopt=False):
+    """
+    :param params: hyperparameters. Its structure is consistent with how search space is defined. See below.
+    :param fpath: Path or URL for the training data used with the model.
+    :param hyperopt: Use hyperopt for hyperparameter search during training.
+    :return: dict with fields 'loss' (scalar loss) and 'status' (success/failure status of run)
+    """
+    max_depth, max_features, n_estimators = params
+    max_depth, max_features, n_estimators = int(max_depth), float(max_features), int(n_estimators)
+
+    df = load_data(fpath)
+
+    X = df.drop(["ArrDelayBinary"], axis=1)
+    y = df["ArrDelayBinary"].astype('int32')
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+    mod = RandomForestClassifier(max_depth=max_depth, max_features=max_features, n_estimators=n_estimators)
+    acc_scorer = accuracy_score
+
+    mod.fit(X_train, y_train)
+    preds = mod.predict(X_test)
+    acc = acc_scorer(y_test, preds)
+
+    mlparams = {"max_depth": str(max_depth),
+                "max_features": str(max_features),
+                "n_estimators": str(n_estimators)}
+    mlflow.log_params(mlparams)
+
+    mlmetrics = {"accuracy": acc}
+    mlflow.log_metrics(mlmetrics)
+
+    if (not hyperopt):
+        return mod
+
+    return {'loss': acc, 'status': STATUS_OK}
+
+
+def train(params, fpath, hyperopt=False):
+    """
+    Proxy function used to call _train
+    :param params: hyperparameters. Its structure is consistent with how search space is defined. See below.
+    :param fpath: Path or URL for the training data used with the model.
+    :param hyperopt: Use hyperopt for hyperparameter search during training.
+    :return: dict with fields 'loss' (scalar loss) and 'status' (success/failure status of run)
+    """
+    with mlflow.start_run(nested=True):
+        return _train(params, fpath, hyperopt)
+
+
+if (__name__ in ("__main__",)):
```

Comment:
No need for parens - looks less like the standard `__name__ == '__main__'` idiom this way